src/Authentication/Common.Authentication/Factories/AuthenticationFactory.cs

```diff
@@ -225,12 +235,26 @@ public ServiceClientCredentials GetServiceClientCredentials(AzureContext context
                 }
                 else if (context.Account.Type == AzureAccount.AccountType.ServicePrincipal)
                 {
-                    result = ApplicationTokenProvider.LoginSilentAsync(
-                        tenant, 
-                        context.Account.Id,
-                        new KeyStoreApplicationCredentialProvider(tenant),
-                        env, 
-                        tokenCache).ConfigureAwait(false).GetAwaiter().GetResult();
+                    var thumbprint = context.Account.GetProperty(AzureAccount.Property.CertificateThumbprint);
```

Comment:
I believe this method throws if the property is not set.  I think we should check IsPropertySet and then get the thumbprint value inside the if body